### ISSUE TITLE
Append the timestamp to the save/replay file name genrate some savefiles for same turn of same scenario

### ISSUE THREAD

**Issue by newfrenchy83** [wesnoth/wesnoth#4936]
### Game and System Information
 - What version of the game are you running? master
 - Where did you download it from? (Steam, Mac App Store, our website, built from source, etc.)built from source
 - If you're using a custom build, what compiler did you use, and what commit did you build at?mingw64 8.1.0
 - What OS are you running?windows 7

### Describe the bug
A clear and concise description of what the bug is. when i reload a scenario in the start, the automatic savegame of turn 1 not erased by new savegame, but i had two savegame of same turn, and if i use return to turn 1, the game chooe then the wrong savefile.

### To Reproduce

1. First.. load a scenario in with start save
2. Then... reload a second time
3. Lastly...

### Expected behavior
A clear and concise description of what you expected to happen. Normaly, each turn must be saved in a unique save file, but the uses of https://github.com/wesnoth/wesnoth/commit/c394353d48d4f1a5db656dcb5a4e5c052cb959a4 genrate a bug

### Savefiles
If applicable, attach savefiles to help explain your problem. (To find savefiles, start Wesnoth, click on the `i` button at the bottom left corner of the title screen, and open the `Saved games` directory.)

### Additional context
Add any other context (compiler logs, console output, etc) about the problem here.

**Comment by Pentarctagon** [comment]
How can Return to Turn 1 load the wrong save file if they're both an automatic start of turn save of the same turn?

**Comment by gfgtdf** [comment]
> How can Return to Turn 1 load the wrong save file if they're both an automatic start of turn save of the same turn?

becasue one of them is from an older playthough, you'd expcet "return to turn X" to load the savefile from the current palythough (which is usually the most recent file)

**Comment by Pentarctagon** [comment]
Alright, so I think there are two separate problems here:
* The back to turn functionality doesn't work at all anymore - it tries to find files by exact filename, which it has no way of knowing anymore because of the timestamp appended to the filename.  @newfrenchy83's problem is most likely caused by him having saves from the same scenario that existed prior to the change I made.
* The autosave functionality creates the turn save per side turn. So previously where it was ending up with only one save per turn because it was overwriting the previous save on each side turn, now each save per side turn is being persisted since, again, the timestamp now makes the filename unique.

**Comment by gfgtdf** [comment]
especially the ""back to turn functionality doesn't work at all anymore" part is quite a problem, if we cannot fins a fix soon i reccomend just to disable the new  functionality for autosaves

**Comment by Pentarctagon** [comment]
The main thing I'm not sure about for that part is how to determine which saves to make available there.  Having the saves be populated at all could be solved by getting all the saves that start with the text that appears before the timestamp part and then order them by the timestamp, but I don't know how I could tell apart different play-throughs of the same scenario.  I wouldn't make any sense to have "Back to..." list
```
Turn 3
Turn 2
Turn 1
Turn 28
Turn 27
...
```
if this is the second time they're playing the scenario.

**Comment by Wedge009** [comment]
So this was caused by #4730? I also recall #4164 being related, which also relates to #1513.

**Comment by Pentarctagon** [comment]
It was caused by the fix I implemented for #4730, to be specific.

**Comment by Pentarctagon** [comment]
I think the best solution, short of reverting the change, is to essentially do the thing that I said didn't make sense.

So essentially change the current logic where it checks for the existence of specific filenames for "Auto-save N", "Auto-save N-1", etc to instead have it look for all saves that start with `<scenario name>` and list the 10(?) most recent ones as
```
Turn 3 Side 2
Turn 3 Side 1
Turn 2 Side 2
Turn 2 Side 1
Turn 1 Side 2
Turn 1 Side 1
...
```
